### PR TITLE
[IMP] condominium: Improving the flows 

### PR DESCRIPTION
**Pull request by dhrs-odoo**
task-4226949

### CODE REVIEW

File: condominium/data/ir_actions_act_window.xml [hunk 1]
```diff
@@ -62,44 +62,116 @@
...
-        <field name="res_model">x_properties</field>
+        <field name="res_model">account.analytic.account</field>
     </record>
     <record id="properties_act_window" model="ir.actions.act_window">
         <field name="help"><![CDATA[
 <p>&nbsp;Create and manage the properties&nbsp;that compose this condominium</p>
 ]]></field>
         <field name="name">Properties</field>
-        <field name="res_model">x_properties</field>
+        <field name="domain">[('plan_id', '=', 2)]</field>
```
Comment (vava-odoo) [inline condominium/data/ir_actions_act_window.xml:120]: what is 2?

File: condominium/data/ir_actions_act_window.xml [hunk 2]
```diff
@@ -62,44 +62,116 @@
...
+                        This is your new action.
+                    </p>
+                    <p>By default, it contains a list and a form view and possibly
+                        other view types depending on the options you chose for your model.
+                    </p>
+                    <p>
+                        You can start customizing these screens by clicking on the Studio icon on the
+                        top right corner (you can also customize this help message there).
+                    </p>
+                ]]></field>
```
Comment (vava-odoo) [inline condominium/data/ir_actions_act_window.xml:135]: useless

File: condominium/data/ir_actions_act_window.xml [hunk 3]
```diff
@@ -62,44 +62,116 @@
...
         <field name="help"><![CDATA[
 <p>&nbsp;Create and manage the properties&nbsp;that compose this condominium</p>
 ]]></field>
         <field name="name">Properties</field>
-        <field name="res_model">x_properties</field>
+        <field name="domain">[('plan_id', '=', 2)]</field>
+        <field name="res_model">account.analytic.account</field>
         <field name="view_mode">kanban,list,form,activity</field>
     </record>
+    <record id="ratios_d3a1aa4d-3ce6-449a-b6fb-c78c0a2de6e5" model="ir.actions.act_window">
```
Comment (vava-odoo) [inline condominium/data/ir_actions_act_window.xml:124]: improve id

File: condominium/data/ir_actions_server.xml [hunk 4]
```diff
@@ -1,22 +1,91 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <odoo>
-    <record id="ir_act_server_split_per_property" model="ir.actions.server">
-        <field name="name">Split per Property</field>
-        <field name="state">code</field>
+    <record id="ir_act_server_split_per_property" model="ir.actions.server" forcecreate="1">
```
Comment (vava-odoo) [inline condominium/data/ir_actions_server.xml:3]: no forcecreate

File: condominium/data/ir_actions_server.xml [hunk 5]
```diff
@@ -1,22 +1,91 @@
...
-    raise UserError('There is no ratio set on the properties. Define a ratio for each property before splitting.')
-for property in record.sudo().partner_id.x_condominium_properties:
-    new_so = record.copy()
-    new_so['x_source_sales_order'] = record.id
-    new_so['partner_id'] = property.x_owner.id
-    new_so['x_property_id'] = property
-    for sol in new_so.order_line:
-        sol['price_unit'] = sol.price_unit * property.x_ratio / total_ratio
-record['state'] = 'cancel'
+properties = env['account.analytic.account'].search([('plan_id', '=', 2), ('id', 'not in', record.x_ratios.x_property.ids)])
```
Comment (vava-odoo) [inline condominium/data/ir_actions_server.xml:52]: 2?

File: condominium/data/ir_default.xml [hunk 6]
```diff
@@ -24,4 +24,8 @@
         <field name="field_id" ref="sequence_properties_types"/>
         <field name="json_value">10</field>
     </record>
+    <record id="ir_default_35" model="ir.default">
+        <field name="field_id" ref="analytic.field_account_analytic_account__plan_id"/>
+        <field name="json_value">2</field>
```
Comment (vava-odoo) [inline condominium/data/ir_default.xml:29]: 2?

File: condominium/data/ir_model.xml [hunk 7]
```diff
@@ -26,4 +26,8 @@
         <field name="model">x_properties_types</field>
         <field name="name">Properties Types</field>
     </record>
+    <record id="ratios_15452286-88ba-4df3-9fa0-41021f013894" model="ir.model">
```
Comment (vava-odoo) [inline condominium/data/ir_model.xml:29]: id

File: condominium/data/ir_model_access.xml [hunk 8]
```diff
@@ -1,5 +1,22 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <odoo>
+    <record id="ratios_group_system_efe40017-03f3-46ec-9551-6ac873674c42" model="ir.model.access">
```
Comment (vava-odoo) [inline condominium/data/ir_model_access.xml:3]: id

File: condominium/data/ir_model_fields.xml [hunk 9]
```diff
@@ -1,505 +1,558 @@
 <?xml version='1.0' encoding='UTF-8'?>
 <odoo>
-    <record id="description_building" model="ir.model.fields">
-        <field name="name">x_name</field>
+    <record id="description_ratios_1292c640-908f-4c8c-a939-20b9b55e6169" model="ir.model.fields">
```
Comment (vava-odoo) [inline condominium/data/ir_model_fields.xml:3]: id

File: condominium/i18n/ar.po
Comment (vava-odoo) [inline condominium/i18n/ar.po:1]: no need to edit .po

**Review by vava-odoo — CHANGES_REQUESTED** [review]
@dhrs-odoo Could you please review yourself with our guidelines first? I listed a few comments but there are many more... Please also avoid useless diff. And runbot is red.

**Comment by dhrs-odoo** [comment]
Hello @vava-odoo 
I have implemented the changes
Please have a look at it once you are free
Thank You :innocent:

File: condominium/data/account_analytic_plan.xml [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="account_analytic_plan_2" model="account.analytic.plan">
+        <field name="name">Properties</field>
+    </record>
+    <record id="account_analytic_plan_3" model="account.analytic.plan">
+        <field name="name">Department</field>
+    </record>
```
Comment (vava-odoo) [inline condominium/data/account_analytic_plan.xml:8]: Why is this one created? I don't see why it is created.

File: condominium/data/ir_actions_act_window.xml [hunk 11]
```diff
@@ -74,32 +92,75 @@
...
         <field name="name">Properties</field>
-        <field name="res_model">x_properties</field>
+        <field name="domain" eval="[('plan_id','=', ref('account_analytic_plan_2'))]"/>
+        <field name="res_model">account.analytic.account</field>
         <field name="view_mode">kanban,list,form,activity</field>
     </record>
+    <record id="ratios_act_window" model="ir.actions.act_window">
+        <field name="name">Ratios</field>
+        <field name="res_model">x_ratios</field>
+      </record>
```
Comment (vava-odoo) [inline condominium/data/ir_actions_act_window.xml:127]:
```suggestion
    </record>
```

File: condominium/data/ir_actions_act_window.xml [hunk 12]
```diff
@@ -74,32 +92,75 @@
...
+                        This is your new action.
+                    </p>
+                    <p>By default, it contains a list and a form view and possibly
+                        other view types depending on the options you chose for your model.
+                    </p>
+                    <p>
+                        You can start customizing these screens by clicking on the Studio icon on the
+                        top right corner (you can also customize this help message there).
+                    </p>
+                ]]></field>
```
Comment (vava-odoo) [inline condominium/data/ir_actions_act_window.xml:155]:
```suggestion
```

File: condominium/data/ir_actions_server.xml [hunk 13]
```diff
@@ -1,22 +1,91 @@
...
+                        'tax_id': sol.tax_id,
+                        'discount': sol.discount,
+                        'analytic_distribution': {analytic_account.id: 100},
+                        }))
+        new_so.write({'order_line': new_so_lines})
+    record['state'] = 'cancel'
+            ]]></field>
         <field name="model_id" ref="sale.model_sale_order"/>
+        <field name="state">code</field>
+        <field name="name">Split per Owner</field>
```
Comment (vava-odoo) [inline condominium/data/ir_actions_server.xml:47]: useless diff

File: condominium/data/ir_model_access.xml [hunk 14]
```diff
@@ -38,7 +55,7 @@
     </record>
     <record id="properties_group_sys" model="ir.model.access">
         <field name="group_id" ref="base.group_system"/>
-        <field name="model_id" ref="model_properties"/>
+        <field name="model_id" ref="analytic.model_account_analytic_account"/>
```
Comment (vava-odoo) [inline condominium/data/ir_model_access.xml:58]: for such a change in stable, it is safer to slightly edit the id and the name

File: condominium/i18n/ar.po
Comment (vava-odoo) [reply condominium/i18n/ar.po:1]: but the pot needs to

File: condominium/data/ir_ui_view.xml [hunk 15]
```diff
@@ -1,429 +1,381 @@
...
+                <field name="x_name" />
+                <field name="x_company_id" groups="base.group_multi_company" />
             </list>
         </field>
     </record>
-    <record id="default_form_view_building" model="ir.ui.view">
-        <field name="model">x_buildings</field>
-        <field name="name">Default form view for x_buildings</field>
-        <field name="type">form</field>
+    <record id="default_form_view_fo_8c2be510-bad6-4e59-9ebc-227f6fd5dfcf" model="ir.ui.view">
```
Comment (vava-odoo) [inline condominium/data/ir_ui_view.xml:68]: :eyes:

File: condominium/data/ir_ui_view.xml [hunk 16]
```diff
@@ -1,429 +1,381 @@
...
+                            <list editable="bottom">
+                                <field name="x_sequence" widget="handle" />
+                                <field name="x_property" domain='[("plan_id", "=", 2)]'/>
+                                <field optional="show" name="x_building" options="{'create_name_field': 'x_name'}" />
+                                <field optional="show" name="x_type" options="{'create_name_field': 'x_name'}" />
+                                <field name="x_owner" optional="show" />
+                                <field optional="show" name="x_floor" />
+                                <field name="x_area" sum="Sum of Area" optional="show" />
+                                <field name="x_ratio" sum="Sum of Ratio" />
+                                <field name="x_company_id" optional="hide" />
```
Comment (vava-odoo) [inline condominium/data/ir_ui_view.xml:47]: don't we need the group multicompany?

File: condominium/data/ir_ui_view.xml [hunk 17]
```diff
@@ -1,429 +1,381 @@
...
-                        <button class="oe_stat_button" icon="fa-home" type="action" name="%(props_build_act_window)d">
-                            <field widget="statinfo" name="x_x_building_x_properties_count" string="Properties"/>
-                        </button>
-                    </div>
-                    <widget name="web_ribbon" text="Archived" bg_color="text-bg-danger" invisible="x_active"/>
-                    <field name="x_image" class="oe_avatar" widget="image"/>
+                <header />
+                <sheet string="Ratios">
+                    <widget name="web_ribbon" text="Archived" bg_color="text-bg-danger" invisible="x_active" />
+                    <field name="x_active" invisible="1" />
```
Comment (vava-odoo) [inline condominium/data/ir_ui_view.xml:78]: Not needed anymore
```suggestion
```

File: condominium/data/ir_ui_view.xml [hunk 18]
```diff
@@ -1,429 +1,381 @@
...
-                                <field name="x_city" placeholder="City" class="o_address_city"/>
-                                <field name="x_state" class="o_address_state" placeholder="State" options="{'no_open': True, 'no_quick_create': True}"/>
-                                <field name="x_zip" placeholder="ZIP" class="o_address_zip"/>
-                                <field name="x_country" placeholder="Country" class="o_address_country" options="{'no_open': True, 'no_create': True}"/>
-                            </div>
-                        </group>
-                        <group>
-                            <field name="x_tag_ids" widget="many2many_tags" options="{'color_field': 'x_color'}"/>
-                            <field name="x_ratio"/>
+                    <group name="studio_group_ae54a3">
```
Comment (vava-odoo) [inline condominium/data/ir_ui_view.xml:84]: You can remove weird names like this one
```suggestion
                    <group>
```

File: condominium/data/ir_ui_view.xml [hunk 19]
```diff
@@ -1,429 +1,381 @@
...
+                        <group name="studio_group_ae54a3_left">
+                            <field name="x_company_id" groups="base.group_multi_company" options="{'no_create': True}" />
                         </group>
-                    </group>
-                    <group>
-                        <field name="x_notes" placeholder="Type down your notes here..." nolabel="1" colspan="2"/>
+                        <group name="studio_group_ae54a3_right" />
```
Comment (vava-odoo) [inline condominium/data/ir_ui_view.xml:88]: useless, no?

File: condominium/data/ir_ui_view.xml [hunk 20]
```diff
@@ -1,429 +1,381 @@
...
             </search>
         </field>
     </record>
-    <record id="default_list_view_building_tags" model="ir.ui.view">
-        <field name="model">x_buildings_tag</field>
-        <field name="name">Default list view for buildings tags</field>
-        <field name="type">list</field>
+    <record id="x_ratios_custom_form_view" model="ir.ui.view">
+        <field name="name">x_ratios.custom.form customization</field>
+        <field name="inherit_id" ref="default_form_view_fo_8c2be510-bad6-4e59-9ebc-227f6fd5dfcf" />
```
Comment (vava-odoo) [inline condominium/data/ir_ui_view.xml:111]: Merge these two views together

File: condominium/data/ir_ui_view.xml [hunk 21]
```diff
@@ -1,429 +1,381 @@
...
+            </data>
         </field>
     </record>
-    <record id="default_list_view_properties" model="ir.ui.view">
-        <field name="name">Default list view for x_properties</field>
-        <field name="model">x_properties</field>
-        <field name="type">list</field>
+    <record id="sale_order_custom_form_view" model="ir.ui.view">
+        <field name="name">sale.order.form customization</field>
+        <field name="inherit_id" ref="sale_subscription.sale_subscription_order_view_form" />
```
Comment (vava-odoo) [inline condominium/data/ir_ui_view.xml:155]: move it back at the top where it was please

File: condominium/data/ir_ui_view.xml [hunk 22]
```diff
@@ -1,429 +1,381 @@
...
+                    </button>
+                </xpath>
+                <xpath expr="//form[1]/sheet[1]/div[not(@name)][1]/h1[1]/field[@name='name']" position="attributes">
+                    <attribute name="string">Apartment...</attribute>
+                </xpath>
+                <xpath expr="//field[@name='active']" position="after">
+                    <xpath expr="//form[1]/sheet[1]/group[@name='main']/group[2]/field[@name='company_id']" position="move" />
+                    <field name="x_condominium" />
+                    <field name="x_building" options='{"create_name_field":"x_name","no_create":true}' required="plan_id == 2" />
+                    <field name="x_type" options="{'create_name_field': 'x_name'}" required="plan_id == 2" />
```
Comment (vava-odoo) [inline condominium/data/ir_ui_view.xml:211]: :eyes:

**Review by vava-odoo — CHANGES_REQUESTED** [review]
Could you please help by reducing the diff, especially the one on fields and views? :pray:

**Review by vava-odoo — APPROVED** [review]

**Comment by vava-odoo** [comment]
@robodoo r+

**Pull request closed without merging**